src/cdp/CdpConnection.ts

```diff
@@ -154,6 +154,22 @@ export class MapperCdpConnection implements CdpConnection {
       const client = this.#sessionCdpClients.get(
         message.sessionId ?? undefined,
       );
+      if (message.method === 'Network.requestWillBeSent') {
+        /**
+         * Emulate CDP event 'Page.frameStartedNavigating' right before the
+         * `Network.requestWillBeSent`.
+         * http://go/webdriver:detect-navigation-started#bookmark=id.64balpqrmadv
+         */
+        const eventParams = message.params;
+        if (eventParams.loaderId === eventParams.requestId) {
+          client?.emit('Page.frameStartedNavigating', {
```

Comment:
that would be visible to external clients right? Let's emit this event on an internal emitter only for the mapper implementation

Reply:
The event will be exposed in `CdpClient` interface and to the clients as `goog:cdp.Page.frameStartedNavigating` event. Are you concerned about the latest, or both?

Reply:
I am concerned that event will be observable by Puppeteer and by clients creating a custom CDP session through Puppeteer who might start depending on it or break due to unknown event/event name format. I do not think even prefixed events should be emitted there. What is the difficulty with using this event internally?

Reply:
Does it even work with custom CDP connections like the one Puppeteer is using?